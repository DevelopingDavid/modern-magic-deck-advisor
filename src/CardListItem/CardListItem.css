.cardlist--item {
  list-style: none;
  margin: 5px 0px 0px 5px
}

.fas,
.far {
  font-size: .5rem;
  padding: 0px 3px;
}