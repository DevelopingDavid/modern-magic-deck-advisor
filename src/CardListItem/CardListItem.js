import React, { Component } from "react";
import '../styles/main.scss';

class CardListItem extends Component {
  removeListItem = () => {
    const [...newUserCardsData] = this.props.userCardsData;
    newUserCardsData.splice(this.props.cardIndex, 1);
    this.props.saveArray('userCardsData', newUserCardsData);
  }

  increaseCardCount = () => {
    if (this.props.cardCount < 4) {
      this.updateUserCardsData(this.props.cardCount + 1);
    }
  }

  decreaseCardCount = () => {
    if (this.props.cardCount > 1) {
      this.updateUserCardsData(this.props.cardCount - 1);
    }
  }

  updateUserCardsData = (newCount) => {
    const newUserCardsData = this.props.userCardsData.map(card => {
      if (this.props.cardName === card.cardName) {
        card.cardCount = newCount;
      }
      return card;
    });
    this.props.saveArray('userCardsData', newUserCardsData);
  }

  render() {
    return (
      <li id={this.props.cardIndex} className="list--li">
        {this.props.cardName}
        <i className="fas fa-minus" onClick={this.decreaseCardCount}/>
        <span> {this.props.cardCount}</span>
        <i className="fas fa-plus" onClick={this.increaseCardCount}/>
        <i className="far fa-trash-alt" onClick={this.removeListItem}/>
      </li>
    );
  }
}

export default CardListItem;