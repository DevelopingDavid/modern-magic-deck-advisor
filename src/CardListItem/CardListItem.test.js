import React from 'react';
import CardListItem from './CardListItem';
import { shallow } from 'enzyme';

const cardName = 'Blood Moon';
const index = 0;
const userCardsData = [{
  cardName: 'Blood Moon',
  cardCount: 1
}];
const removeListItemMock = jest.fn();
const setCardCountMock = jest.fn();

describe('CardListItem', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CardListItem 
        cardName={cardName}
        cardIndex={index}
        key={cardName}
        removeListItem={removeListItemMock}
        setCardCount={setCardCountMock} 
        userCardsData={userCardsData}/>
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call removeListItem when the trash-can is clicked', () => {
    wrapper.find('.cardlist--item').simulate('click', {
      target: {
        classList: {
          contains: (className) => {
            if (className === 'fa-trash-alt') {
              return true;
            } else {
              return false;
            }
          }
        },
        closest: () => {
          return {id: '0'}
        }
      }
    })
    expect(removeListItemMock).toBeCalled();
  })

})