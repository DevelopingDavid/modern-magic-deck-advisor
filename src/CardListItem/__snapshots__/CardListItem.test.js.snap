// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardListItem should match the snapshot 1`] = `
<li
  className="list--li"
  id={0}
>
  Blood Moon
  <i
    className="fas fa-minus"
    onClick={[Function]}
  />
  <span>
     
    2
  </span>
  <i
    className="fas fa-plus"
    onClick={[Function]}
  />
  <i
    className="far fa-trash-alt"
    onClick={[Function]}
  />
</li>
`;
